RTSP publish causing panic/runtime error

## Which version are you using?

v1.1.0

## Which operating system are you using?

- [x] Linux amd64 standard
- [ ] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue

Error publishing an RTSP stream to v1.1.0 `panic: runtime error: index out of range [0] with length 0`. This same video source has been used with mediamtx v1.0.3 and back to `rtsp-simple-server` and always worked fine

## Describe how to replicate the issue

1. start the server
2. publish with ffmpeg
3. observe the logs 

## Did you attach the server logs?

yes

```
2023/09/20 03:12:32 INF [RTSP] [session 4056da8d] is publishing to path '78067a492b07293b4467579110533cdc', 2 tracks (H265, MPEG-4 Audio)
panic: runtime error: index out of range [0] with length 0
compass-rtsp-server_1  |
goroutine 80 [running]:
github.com/bluenviron/mediamtx/internal/formatprocessor.rtpH265ExtractVPSSPSPPS({0xc000326ccc?, 0xc000514a58?, 0x4103a5?})
       /build/mediamtx/internal/formatprocessor/h265.go:58 +0x248
github.com/bluenviron/mediamtx/internal/formatprocessor.(*formatProcessorH265).updateTrackParametersFromRTPPacket(0xc0003806a0, {0xc000326ccc?, 0x0?, 0x0?})
       /build/mediamtx/internal/formatprocessor/h265.go:122 +0x38
github.com/bluenviron/mediamtx/internal/formatprocessor.(*formatProcessorH265).ProcessRTPPacket(0xc0003806a0, 0xc00017f950, {0xc000514c38?, 0x411784?, 0x1681de0?}, 0x1e8f508b, 0x0)
       /build/mediamtx/internal/formatprocessor/h265.go:271 +0x145
github.com/bluenviron/mediamtx/internal/stream.(*streamFormat).writeRTPPacket(0xc0000192c0, 0xc000019260?, 0xc0000029c0?, 0x1e8f508b?, {0x15f90?, 0xad08b?, 0x1681de0?}, 0xa02d20?)
       /build/mediamtx/internal/stream/stream_format.go:78 +0x63
github.com/bluenviron/mediamtx/internal/stream.(*Stream).WriteRTPPacket(0xc00070ca40, 0xc0000029c0, {0xfff2b8, 0xc0001c20e0}, 0xc0003be028?, {0x1681de0?, 0x0?, 0x1681de0?}, 0xa03a40?)
       /build/mediamtx/internal/stream/stream.go:147 +0x12f
github.com/bluenviron/mediamtx/internal/core.(*rtspSession).onRecord.func1(0xc0003be000?)
       /build/mediamtx/internal/core/rtsp_session.go:370 +0x88
github.com/bluenviron/gortsplib/v4.(*serverSessionFormat).readRTPTCP(0xc00070c880, 0xc00017f950)
       /go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.1.0/server_session_format.go:98 +0x130
github.com/bluenviron/gortsplib/v4.(*serverSessionMedia).readRTPTCPRecord(0xc000488a80, {0xc000326cc0, 0x5d, 0x5d})
       /go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.1.0/server_session_media.go:285 +0x85
github.com/bluenviron/gortsplib/v4.(*serverConnReader).readFuncTCP(0xc0000358e0)
       /go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.1.0/server_conn_reader.go:126 +0x315
github.com/bluenviron/gortsplib/v4.(*serverConnReader).run(0xc0000358e0)
       /go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.1.0/server_conn_reader.go:51 +0x84
created by github.com/bluenviron/gortsplib/v4.newServerConnReader in goroutine 79
       /go/pkg/mod/github.com/bluenviron/gortsplib/v4@v4.1.0/server_conn_reader.go:36 +0xa9
```

## Did you attach a network dump?

no

--- a/internal/formatprocessor/h264_test.go
+++ b/internal/formatprocessor/h264_test.go
@@ -14,53 +14,68 @@ import (
 )
 
 func TestH264DynamicParams(t *testing.T) {
-	forma := &format.H264{
-		PayloadTyp:        96,
-		PacketizationMode: 1,
+	for _, ca := range []string{"standard", "aggregated"} {
+		t.Run(ca, func(t *testing.T) {
+			forma := &format.H264{
+				PayloadTyp:        96,
+				PacketizationMode: 1,
+			}
+
+			p, err := New(1472, forma, false)
+			require.NoError(t, err)
+
+			enc, err := forma.CreateEncoder()
+			require.NoError(t, err)
+
+			pkts, err := enc.Encode([][]byte{{byte(h264.NALUTypeIDR)}})
+			require.NoError(t, err)
+
+			data, err := p.ProcessRTPPacket(pkts[0], time.Time{}, 0, true)
+			require.NoError(t, err)
+
+			require.Equal(t, [][]byte{
+				{byte(h264.NALUTypeIDR)},
+			}, data.(*unit.H264).AU)
+
+			if ca == "standard" {
+				pkts, err = enc.Encode([][]byte{{7, 4, 5, 6}}) // SPS
+				require.NoError(t, err)
+
+				_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
+				require.NoError(t, err)
+
+				pkts, err = enc.Encode([][]byte{{8, 1}}) // PPS
+				require.NoError(t, err)
+
+				_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
+				require.NoError(t, err)
+			} else {
+				pkts, err = enc.Encode([][]byte{
+					{7, 4, 5, 6}, // SPS
+					{8, 1},       // PPS
+				})
+				require.NoError(t, err)
+
+				_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
+				require.NoError(t, err)
+			}
+
+			require.Equal(t, []byte{7, 4, 5, 6}, forma.SPS)
+			require.Equal(t, []byte{8, 1}, forma.PPS)
+
+			pkts, err = enc.Encode([][]byte{{byte(h264.NALUTypeIDR)}})
+			require.NoError(t, err)
+
+			data, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, true)
+			require.NoError(t, err)
+
+			require.Equal(t, [][]byte{
+				{0x07, 4, 5, 6},
+				{0x08, 1},
+				{byte(h264.NALUTypeIDR)},
+			}, data.(*unit.H264).AU)
+		})
 	}
-
-	p, err := New(1472, forma, false)
-	require.NoError(t, err)
-
-	enc, err := forma.CreateEncoder()
-	require.NoError(t, err)
-
-	pkts, err := enc.Encode([][]byte{{byte(h264.NALUTypeIDR)}})
-	require.NoError(t, err)
-
-	data, err := p.ProcessRTPPacket(pkts[0], time.Time{}, 0, true)
-	require.NoError(t, err)
-
-	require.Equal(t, [][]byte{
-		{byte(h264.NALUTypeIDR)},
-	}, data.(*unit.H264).AU)
-
-	pkts, err = enc.Encode([][]byte{{7, 4, 5, 6}}) // SPS
-	require.NoError(t, err)
-
-	_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
-	require.NoError(t, err)
-
-	pkts, err = enc.Encode([][]byte{{8, 1}}) // PPS
-	require.NoError(t, err)
-
-	_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
-	require.NoError(t, err)
-
-	require.Equal(t, []byte{7, 4, 5, 6}, forma.SPS)
-	require.Equal(t, []byte{8, 1}, forma.PPS)
-
-	pkts, err = enc.Encode([][]byte{{byte(h264.NALUTypeIDR)}})
-	require.NoError(t, err)
-
-	data, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, true)
-	require.NoError(t, err)
-
-	require.Equal(t, [][]byte{
-		{0x07, 4, 5, 6},
-		{0x08, 1},
-		{byte(h264.NALUTypeIDR)},
-	}, data.(*unit.H264).AU)
 }
 
 func TestH264OversizedPackets(t *testing.T) {

--- a/internal/formatprocessor/h265.go
+++ b/internal/formatprocessor/h265.go
@@ -55,7 +55,7 @@ func rtpH265ExtractVPSSPSPPS(payload []byte) ([]byte, []byte, []byte) {
 			nalu := payload[:size]
 			payload = payload[size:]
 
-			typ = h265.NALUType((payload[0] >> 1) & 0b111111)
+			typ = h265.NALUType((nalu[0] >> 1) & 0b111111)
 
 			switch typ {
 			case h265.NALUType_VPS_NUT:

--- a/internal/formatprocessor/h265_test.go
+++ b/internal/formatprocessor/h265_test.go
@@ -14,60 +14,76 @@ import (
 )
 
 func TestH265DynamicParams(t *testing.T) {
-	forma := &format.H265{
-		PayloadTyp: 96,
+	for _, ca := range []string{"standard", "aggregated"} {
+		t.Run(ca, func(t *testing.T) {
+			forma := &format.H265{
+				PayloadTyp: 96,
+			}
+
+			p, err := New(1472, forma, false)
+			require.NoError(t, err)
+
+			enc, err := forma.CreateEncoder()
+			require.NoError(t, err)
+
+			pkts, err := enc.Encode([][]byte{{byte(h265.NALUType_CRA_NUT) << 1, 0}})
+			require.NoError(t, err)
+
+			data, err := p.ProcessRTPPacket(pkts[0], time.Time{}, 0, true)
+			require.NoError(t, err)
+
+			require.Equal(t, [][]byte{
+				{byte(h265.NALUType_CRA_NUT) << 1, 0},
+			}, data.(*unit.H265).AU)
+
+			if ca == "standard" {
+				pkts, err = enc.Encode([][]byte{{byte(h265.NALUType_VPS_NUT) << 1, 1, 2, 3}})
+				require.NoError(t, err)
+
+				_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
+				require.NoError(t, err)
+
+				pkts, err = enc.Encode([][]byte{{byte(h265.NALUType_SPS_NUT) << 1, 4, 5, 6}})
+				require.NoError(t, err)
+
+				_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
+				require.NoError(t, err)
+
+				pkts, err = enc.Encode([][]byte{{byte(h265.NALUType_PPS_NUT) << 1, 7, 8, 9}})
+				require.NoError(t, err)
+
+				_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
+				require.NoError(t, err)
+			} else {
+				pkts, err = enc.Encode([][]byte{
+					{byte(h265.NALUType_VPS_NUT) << 1, 1, 2, 3},
+					{byte(h265.NALUType_SPS_NUT) << 1, 4, 5, 6},
+					{byte(h265.NALUType_PPS_NUT) << 1, 7, 8, 9},
+				})
+				require.NoError(t, err)
+
+				_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
+				require.NoError(t, err)
+			}
+
+			require.Equal(t, []byte{byte(h265.NALUType_VPS_NUT) << 1, 1, 2, 3}, forma.VPS)
+			require.Equal(t, []byte{byte(h265.NALUType_SPS_NUT) << 1, 4, 5, 6}, forma.SPS)
+			require.Equal(t, []byte{byte(h265.NALUType_PPS_NUT) << 1, 7, 8, 9}, forma.PPS)
+
+			pkts, err = enc.Encode([][]byte{{byte(h265.NALUType_CRA_NUT) << 1, 0}})
+			require.NoError(t, err)
+
+			data, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, true)
+			require.NoError(t, err)
+
+			require.Equal(t, [][]byte{
+				{byte(h265.NALUType_VPS_NUT) << 1, 1, 2, 3},
+				{byte(h265.NALUType_SPS_NUT) << 1, 4, 5, 6},
+				{byte(h265.NALUType_PPS_NUT) << 1, 7, 8, 9},
+				{byte(h265.NALUType_CRA_NUT) << 1, 0},
+			}, data.(*unit.H265).AU)
+		})
 	}
-
-	p, err := New(1472, forma, false)
-	require.NoError(t, err)
-
-	enc, err := forma.CreateEncoder()
-	require.NoError(t, err)
-
-	pkts, err := enc.Encode([][]byte{{byte(h265.NALUType_CRA_NUT) << 1, 0}})
-	require.NoError(t, err)
-
-	data, err := p.ProcessRTPPacket(pkts[0], time.Time{}, 0, true)
-	require.NoError(t, err)
-
-	require.Equal(t, [][]byte{
-		{byte(h265.NALUType_CRA_NUT) << 1, 0},
-	}, data.(*unit.H265).AU)
-
-	pkts, err = enc.Encode([][]byte{{byte(h265.NALUType_VPS_NUT) << 1, 1, 2, 3}})
-	require.NoError(t, err)
-
-	_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
-	require.NoError(t, err)
-
-	pkts, err = enc.Encode([][]byte{{byte(h265.NALUType_SPS_NUT) << 1, 4, 5, 6}})
-	require.NoError(t, err)
-
-	_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
-	require.NoError(t, err)
-
-	pkts, err = enc.Encode([][]byte{{byte(h265.NALUType_PPS_NUT) << 1, 7, 8, 9}})
-	require.NoError(t, err)
-
-	_, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, false)
-	require.NoError(t, err)
-
-	require.Equal(t, []byte{byte(h265.NALUType_VPS_NUT) << 1, 1, 2, 3}, forma.VPS)
-	require.Equal(t, []byte{byte(h265.NALUType_SPS_NUT) << 1, 4, 5, 6}, forma.SPS)
-	require.Equal(t, []byte{byte(h265.NALUType_PPS_NUT) << 1, 7, 8, 9}, forma.PPS)
-
-	pkts, err = enc.Encode([][]byte{{byte(h265.NALUType_CRA_NUT) << 1, 0}})
-	require.NoError(t, err)
-
-	data, err = p.ProcessRTPPacket(pkts[0], time.Time{}, 0, true)
-	require.NoError(t, err)
-
-	require.Equal(t, [][]byte{
-		{byte(h265.NALUType_VPS_NUT) << 1, 1, 2, 3},
-		{byte(h265.NALUType_SPS_NUT) << 1, 4, 5, 6},
-		{byte(h265.NALUType_PPS_NUT) << 1, 7, 8, 9},
-		{byte(h265.NALUType_CRA_NUT) << 1, 0},
-	}, data.(*unit.H265).AU)
 }
 
 func TestH265OversizedPackets(t *testing.T) {

